### ISSUE TITLE
.intentdefinition file defaults to resource not sources buildPhase attribute

### ISSUE THREAD

**Issue by DanEdgarTarget** [yonaskolb/XcodeGen#439]
While adding an `.intentdefinition` file, I noticed that Xcodegen includes the `.intentdefinition` file as a resource not as a source file.

This causes the normal Xcode compile time code generation which triggers from the `.intentdefinition` to not actually do the codegen for the `.intentdefinition` file so none of the required SiriKit Intent based classes defined in the `.intentdefinition` file appear in the target.

The workaround is just to explicitly set the `buildPhase` to `sources` for the `.intentdefinition` file:
```
sources:
      - path: App/Sources/Application/Routing/Intents.intentdefinition
        buildPhase: sources
 ...
```

It would be good if the default for Xcodegen `.intentdefinition` file type inclusion set the `buildPhase` to `sources` automatically so the manual file include wouldn't be necessary.

**Comment by yonaskolb** [comment]
Hi @DanEdgarTarget, good find. Those files are indeed missing the correct build phase.
Would you like to open a PR to fix this? It's a simple change in a single file here https://github.com/yonaskolb/XcodeGen/blob/master/Sources/XcodeGenKit/SourceGenerator.swift#L186
The `source` case just needs that `intentdefinition`.

Github makes it really easy to create simple PR's like this. You can click `edit` on that file and make the change, and then it will prompt you to create a commit, branch, and finally a PR.

**Comment by yonaskolb** [comment]
Fixed in #442. Will be the next release

**Comment by cremnet** [comment]
I have the same problem, but with localized .intentdefinition
It works only when path is directly to .intentdefinition but localisation is broken in that case
